src/my_unix.ml

```diff
@@ -58,17 +58,18 @@ let at_exit_once callback =
   end
 
 let run_and_open s kont =
-  let ic =
+  let ic, cleanup =
     if Sys.win32
     then
-      let args = My_std.prepare_command_for_windows s in
-      Unix.open_process_args_in args.(0) args
-    else Unix.open_process_in s in
+      let args, cleanup = My_std.prepare_command_for_windows s in
+      Unix.open_process_args_in args.(0) args, cleanup
+    else Unix.open_process_in s, None in
   let close () =
     match Unix.close_process_in ic with
-    | Unix.WEXITED 0 -> ()
+    | Unix.WEXITED 0 -> Option.iter (fun f -> f ()) cleanup
     | Unix.WEXITED _ | Unix.WSIGNALED _ | Unix.WSTOPPED _ ->
-        failwith (Printf.sprintf "Error while running: %s" s) in
+      Option.iter (fun f -> f ()) cleanup;
+      failwith (Printf.sprintf "Error while running: %s" s) in
```

Comment:
I wonder if we would benefit from splitting the two paths more.

Untested:
```ocaml
let close ic = ... in
if not Sys.win32 then begin
  let ic = Unix.open_process_in s in
  Fun.protect ~finally:(fun () -> close ic) @@ fun () ->
  kont ic
else
  let args, cleanup = My_std.prepare_command_for_windows s in
  let ic = Unix.open_process_args_in args.(0) args in
  Fun.protect ~finally:cleanup @@ fun () ->
  Fun.protect ~finally:(fun () -> close ic) @@ fun () ->
  kont ic
```

Reply:
(Or whatever style other than `@@ fun () ->` that you prefer or find more consistent with the codebase.)

Reply:
done

Reply:
This patch makes the CI complain because it changes the exception raise:
- previously: `Failure ...`
- with this patch: `Finally_raised ..`

I'll revert that part